plugins/zenoh-plugin-rest/src/lib.rs

```diff
@@ -521,48 +417,140 @@ pub async fn run(runtime: DynamicRuntime, conf: Config) -> ZResult<()> {
     // But cannot be done twice in case of static link.
     zenoh::init_log_from_env_or("error");
 
-    let zid = runtime.zid().to_string();
-    let session = zenoh::session::init(runtime).await.unwrap();
-
-    let mut app = Server::with_state((Arc::new(session), zid));
-    app.with(
-        tide::security::CorsMiddleware::new()
-            .allow_methods(
-                "GET, POST, PUT, PATCH, DELETE"
-                    .parse::<http_types::headers::HeaderValue>()
-                    .unwrap(),
-            )
-            .allow_origin(tide::security::Origin::from("*"))
-            .allow_credentials(false),
-    );
-
-    app.at("/")
-        .get(query)
-        .post(query)
-        .put(write)
-        .patch(write)
-        .delete(write);
-    app.at("*")
-        .get(query)
-        .post(query)
-        .put(write)
-        .patch(write)
-        .delete(write);
-
-    if let Err(e) = app.listen(conf.http_port).await {
-        tracing::error!("Unable to start http server for REST: {:?}", e);
-        return Err(e.into());
+    let zid = runtime.zid();
+    let session = zenoh::session::init(runtime).await?;
+
+    let app = Router::new()
+        .route(
+            "/{*key_expr}",
+            get(subscribe_or_query)
```

Comment:
Does it miss the POST?

Reply:
https://zenoh.io/docs/apis/rest/
There is no POST in the official documentation, and POST to make query or subscribe makes no sense.